### PR TITLE
JAMES-3056 renaming mailboxes doesn't need ACL to be involved

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java [hunk 1]
```diff
@@ -179,8 +178,9 @@ public MailboxId save(Mailbox mailbox) throws MailboxException {
     private boolean trySave(Mailbox cassandraMailbox, CassandraId cassandraId) {
         boolean isCreated = mailboxPathV2DAO.save(cassandraMailbox.generateAssociatedPath(), cassandraId).block();
         if (isCreated) {
-            Optional<Mailbox> simpleMailbox = retrieveMailbox(cassandraId).blockOptional();
-            simpleMailbox.ifPresent(mbx -> mailboxPathV2DAO.delete(mbx.generateAssociatedPath()).block());
+            mailboxDAO.retrieveMailbox(cassandraId)
+                .blockOptional()
```
Comment (chibenwa) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java]: Can we try to reactor-chain this?

**Comment by trantienduchn** [comment]
This probably will cause a very bad conflict :)

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java [hunk 2]
```diff
@@ -176,14 +176,14 @@ public MailboxId save(Mailbox mailbox) throws MailboxException {
...
-                .ifPresent(mbx -> mailboxPathV2DAO.delete(mbx.generateAssociatedPath()).block());
-            mailboxDAO.save(cassandraMailbox).block();
-        }
-        return isCreated;
+        return mailboxPathV2DAO.save(cassandraMailbox.generateAssociatedPath(), cassandraId)
+            .filter(isCreated -> isCreated)
+            .flatMap(mailboxHasCreated -> mailboxDAO.retrieveMailbox(cassandraId)
+                .flatMap(mailbox -> mailboxPathV2DAO.delete(mailbox.generateAssociatedPath()))
+                .then(mailboxDAO.save(cassandraMailbox))
+                .then(Mono.just(mailboxHasCreated)))
```
Comment (chibenwa) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java]: `then(Mono.just(mailboxHasCreated)))` -> `.thenReturn(mailboxHasCreated)`

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java [hunk 3]
```diff
@@ -176,14 +176,14 @@ public MailboxId save(Mailbox mailbox) throws MailboxException {
...
-                .blockOptional()
-                .ifPresent(mbx -> mailboxPathV2DAO.delete(mbx.generateAssociatedPath()).block());
-            mailboxDAO.save(cassandraMailbox).block();
-        }
-        return isCreated;
+        return mailboxPathV2DAO.save(cassandraMailbox.generateAssociatedPath(), cassandraId)
+            .filter(isCreated -> isCreated)
+            .flatMap(mailboxHasCreated -> mailboxDAO.retrieveMailbox(cassandraId)
+                .flatMap(mailbox -> mailboxPathV2DAO.delete(mailbox.generateAssociatedPath()))
+                .then(mailboxDAO.save(cassandraMailbox))
```
Comment (chibenwa) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java]: This save SHOULD always be performed, even if `mailboxDAO.retrieveMailbox(cassandraId)` returned empty

**Comment by Arsnael** [comment]
tests are :green_apple: 
```
[e666b2a0c8b0a78800db9d380ad99db33542799a] [ERROR] Errors: 
[e666b2a0c8b0a78800db9d380ad99db33542799a] [ERROR]   JamesDeploymentValidationTest>DeploymentValidation.validateDeploymentWithMailsFromSmtp:95 » ConditionTimeout
```
test this please

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java (hunk 2 above)
Comment (mbaechler) [reply mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java]: it's actually `thenReturn(true)`

**Comment by trantienduchn** [comment]
> I don't understand what this PR is doing: I see no behavior change in the production code and a lot of removed tests. Could you elaborate about why?

It's a little optimization, the old method `retrieveMailbox()` requires a query to MailboxACL table. It's not necessary for creating or & renaming a mailbox. There's no behavior change, just optimization.

Before that, I wrote some tests that setup the ACL to return Mono.error() on some case of creating & renaming mailbox. Since ACL is not required anymore, I removed those test because they are white box tests, I know ACL doesn't involve, so keeping them adds no value for me.

**Comment by Arsnael** [comment]
```
[cdb4c1abcca962aa3aaa767057b6442e8283ed9d] Exception in thread "Thread-3136" java.lang.RuntimeException: java.io.IOException: Failed to bind to /0.0.0.0:36595
[cdb4c1abcca962aa3aaa767057b6442e8283ed9d] 	at org.apache.james.webadmin.WebAdminServer.lambda$start$1(WebAdminServer.java:95)
[cdb4c1abcca962aa3aaa767057b6442e8283ed9d] 	at spark.Service.lambda$init$2(Service.java:582)
[cdb4c1abcca962aa3aaa767057b6442e8283ed9d] 	at java.base/java.lang.Thread.run(Thread.java:834)
[cdb4c1abcca962aa3aaa767057b6442e8283ed9d] Caused by: java.io.IOException: Failed to bind to /0.0.0.0:36595
```
test this please

**Comment by Arsnael** [comment]
```
[cdb4c1abcca962aa3aaa767057b6442e8283ed9d] [ERROR] Errors: 
[cdb4c1abcca962aa3aaa767057b6442e8283ed9d] [ERROR]   AliasMappingTest.userShouldNotReceiveDuplicatesWhenUserAndAliasRegisteredToAGroup:417 » ConditionTimeout
```
test this please

**Comment by trantienduchn** [comment]
test this please

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**